src/garage/sampler/base.py

```diff
@@ -1,43 +1,79 @@
+"""Base sampler class."""
+
 import abc
+import copy
 
 
 class Sampler(abc.ABC):
-    @abc.abstractmethod
+    """Base class of all samplers."""
```

Comment:
Definitely a TODO here is to describe the sampler design in a little more detail